### PR TITLE
Started work

### PR DESCRIPTION
**Pull request by ami-friedman**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,162 +1,12 @@
-# Byte-compiled / optimized / DLL files
+.idea/
 __pycache__/
-*.py[cod]
-*$py.class
+jjbroker.log
```
Comment (gs202) [inline .gitignore]: lol

File: app/core/db.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from sqlmodel import create_engine
+
+from app.core.config import settings
+from app.core.models import ExpenseBudget  # noqa: F401
+from app.core.models import ExpenseCategory  # noqa: F401
+from app.core.models import ExpenseCategoryGroup  # noqa: F401
+from app.core.models import ExpenseTransaction  # noqa: F401
```
Comment (gs202) [inline app/core/db.py]: why do you import those modules. im sure there is another way to ensure all tables exist without this.
consider changing the SQL you are using to be less "maverick"

File: app/core/models.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from datetime import date
+
+from sqlmodel import Field
+from sqlmodel import SQLModel
+
+
+class ExpenseCategoryGroup(SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
```
Comment (gs202) [inline app/core/models.py]: better use Optional, its a union of x and None

File: app/core/models.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+class ExpenseBudget(SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    month: date = Field(default=None, nullable=False)
+    category_id: int = Field(default=None, nullable=False, foreign_key='expensecategory.id')
+    category_group_id: int = Field(default=None, nullable=False, foreign_key='expensecategorygroup.id')
+    amount: float = Field(default=None, nullable=False)
+
+    class Config:
+        arbitrary_types_allowed = True
```
Comment (gs202) [inline app/core/models.py]: what's the reason for having this?

File: app/core/db.py (hunk 2 above)
Comment (ami-friedman) [reply app/core/db.py]: What do you mean by less maverick?

I was basically following the pattern from Fast API's full stack template:
https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/core/db.py

File: app/core/models.py (hunk 3 above)
Comment (ami-friedman) [reply app/core/models.py]: So yea this what I saw being used in that template. I thought "Optional" was more the old Python type-hint style, no?

File: app/core/models.py (hunk 4 above)
Comment (ami-friedman) [reply app/core/models.py]: In order for "date" to be supported as a type I had to add this

File: app/core/db.py (hunk 2 above)
Comment (gs202) [reply app/core/db.py]: Check sqlalchemy, maybe it will make things easier (something simmilar to Orion)

File: app/core/models.py (hunk 3 above)
Comment (gs202) [reply app/core/models.py]: Both are fine. i guess that you are right here

File: app/core/models.py (hunk 4 above)
Comment (gs202) [reply app/core/models.py]: maybe use datetime instead? AFAIK it will work without this

**Pull request merged**